Source/SwiftLintFramework/Rules/Metrics/EnumCaseAssociatedValuesLengthRule.swift

```diff
@@ -0,0 +1,74 @@
+import SourceKittenFramework
+
+public struct EnumCaseAssociatedValuesLengthRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityLevelsConfiguration(warning: 5, error: 6)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "enum_case_associated_values_count",
+        name: "Enum Case Associated Values Count",
+        description: "Number of associated values should be low",
+        kind: .metrics,
+        nonTriggeringExamples: [
+            """
+            enum Employee {
+                case fullTime(name: String, retirement: Date, designation: String, contactNumber: Int)
+                case partTime(name: String, age: Int, contractEndDate: Date)
+            }
+            """
+        ],
+        triggeringExamples: [
+            """
+            enum Employee {
+                case fullTime(name: String, retirement: Date, age: Int, designation: String, contactNumber: Int)
```

Comment:
Can you please add the violation mark (`↓`) in the examples so we can test that the violation is triggered in the expected location?

It'd be great to also add examples with unnamed associated values (e.g. `bar(String)`)